--- a/Makefile
+++ b/Makefile
@@ -52,8 +52,8 @@
 	cp backends/initramfs-tools/live.script $(DESTDIR)/usr/share/initramfs-tools/scripts/live
 
 	# Installing docs
-	mkdir -p $(DESTDIR)/usr/share/doc/live-boot
-	cp -r COPYING $(DESTDIR)/usr/share/doc/live-boot
+	mkdir -p $(DESTDIR)/usr/share/doc/live-boot-grml
+	cp -r COPYING $(DESTDIR)/usr/share/doc/live-boot-grml
 
 	# Installing manpages
 	for MANPAGE in manpages/en/*; \
@@ -86,7 +86,7 @@
 	rmdir --ignore-fail-on-non-empty $(DESTDIR)/usr > /dev/null 2>&1 || true
 
 	# Uninstalling docs
-	rm -rf $(DESTDIR)/usr/share/doc/live-boot
+	rm -rf $(DESTDIR)/usr/share/doc/live-boot-grml
 	rmdir --ignore-fail-on-non-empty $(DESTDIR)/usr/share/doc > /dev/null 2>&1 || true
 	rmdir --ignore-fail-on-non-empty $(DESTDIR)/usr/share > /dev/null 2>&1 || true
 	rmdir --ignore-fail-on-non-empty $(DESTDIR)/usr > /dev/null 2>&1 || true
