Index: live-boot-grml/hooks/kms
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ live-boot-grml/backends/initramfs-tools/kms.hook	2011-12-06 22:09:55.000000000 +0100
@@ -0,0 +1,25 @@
+#!/bin/sh
+# hook for adding kms modules
+
+set -e
+
+# initramfs-tools header
+
+PREREQ=""
+prereqs()
+{
+	echo "${PREREQ}"
+}
+
+case "${1}" in
+	prereqs)
+		prereqs
+		exit 0
+		;;
+esac
+
+. /usr/share/initramfs-tools/hook-functions
+
+for module in i915 radeon nouveau ; do
+	manual_add_modules $module
+done
