Index: live-boot-grml/components/9990-misc-helpers.sh
===================================================================
--- live-boot-grml.orig/components/9990-misc-helpers.sh	2014-03-08 13:38:26.286476006 +0100
+++ live-boot-grml/components/9990-misc-helpers.sh	2014-03-08 13:38:26.282476475 +0100
@@ -443,6 +443,12 @@
 		return 1
 	fi
 
+	# get_fstype might report "unknown" or "swap", ignore it as no such kernel module exists
+	if [ "${fstype}" = "unknown" ] || [ "${fstype}" = "swap" ]
+	then
+		return 1
+	fi
+
 	# Try to look if it is already supported by the kernel
 	if grep -q ${fstype} /proc/filesystems
 	then
