--- a/scripts/boot/9990-grml-networking.sh
+++ b/scripts/boot/9990-grml-networking.sh
@@ -59,6 +59,14 @@
         method="manual"
     fi
 
+    # if boot option "nodhcp" is set but also boot option "dhcp" is
+    # set, then dhcp should win over it as we default to dhcp and if
+    # nodhcp is used as default boot option but "dhcp" is added then it
+    # would be confusing to not get a working network setup
+    if [ "$DHCP" = "true" ] ; then
+        method="dhcp"
+    fi
+
     cat >> $IFFILE << EOF
 allow-hotplug ${interface}
 iface ${interface} inet ${method}
