--- a/scripts/boot/0020-read-only
+++ b/scripts/boot/0020-read-only
@@ -53,4 +53,20 @@
 				;;
 		esac
 	done
+
+	if grep -qe persistence /proc/cmdline
+		then
+		printf " * Persistence mode enabled, searching for persistency related devices to unlock\n" >/dev/console
+
+		for label in custom-ov home-rw home-sn live-rw live-sn persistence
+		do
+			if blkid -t LABEL="$label" | grep -q '.'
+			then
+				device=$(blkid -t LABEL="$label" | awk -F: '{print $1}')
+				printf "   - Setting device %-9s with label '%s' to write mode for persistence mode: " "$device" "$label" >/dev/console
+				blockdev --setrw $device && printf "done\n" >/dev/console || printf "failed\n" >/dev/console
+			fi
+		done
+	fi
+
 }
