--- a/Makefile
+++ b/Makefile
@@ -47,9 +47,13 @@
 
 	# Installing executables
 	mkdir -p $(DESTDIR)/usr/share/initramfs-tools/hooks
-	cp backends/initramfs-tools/live.hook $(DESTDIR)/usr/share/initramfs-tools/hooks/live
+	for f in backends/initramfs-tools/*.hook; do \
+		cp $${f} $(DESTDIR)/usr/share/initramfs-tools/hooks/$$(basename $${f} .hook); \
+	done
 	mkdir -p $(DESTDIR)/usr/share/initramfs-tools/scripts
-	cp backends/initramfs-tools/live.script $(DESTDIR)/usr/share/initramfs-tools/scripts/live
+	for f in backends/initramfs-tools/*.script; do \
+		cp $${f} $(DESTDIR)/usr/share/initramfs-tools/scripts/$$(basename $${f} .script); \
+	done
 
 	# Installing docs
 	mkdir -p $(DESTDIR)/usr/share/doc/live-boot-grml
