From 135a17956c50102b978d34d0779c354bb08e9cf9 Mon Sep 17 00:00:00 2001
From: Evgeni Golov <evgeni@debian.org>
Date: Mon, 17 Dec 2012 16:52:25 +0100
Subject: [PATCH] do not fix mountpoints with live-persistence.conf as home-rw/live-rw

--- a/scripts/boot/9990-misc-helpers.sh
+++ b/scripts/boot/9990-misc-helpers.sh
@@ -1610,7 +1610,7 @@
 	fi
 
 	include_list="${backing}/${persistence_list}"
-	if [ ! -r "${include_list}" ]
+	if [ ! -r "${include_list}" ] && [ ! -r "${backing}/${old_persistence_list}" ]
 	then
 		echo "# persistence backwards compatibility:
 ${dir} ${opt},source=." > "${include_list}"
