--- a/scripts/boot/9990-misc-helpers.sh
+++ b/scripts/boot/9990-misc-helpers.sh
@@ -202,7 +202,7 @@
 				# Adding lvm support
 				if [ -x /scripts/local-top/lvm2 ]
 				then
-					ROOT="$device" resume="" /scripts/local-top/lvm2
+					ROOT="$device" resume="" /scripts/local-top/lvm2 >>/boot.log
 				fi
 				;;
 
@@ -210,10 +210,10 @@
 				# Adding raid support
 				if [ -x /scripts/local-top/mdadm ]
 				then
-					cp /conf/conf.d/md /conf/conf.d/md.orig
+					[ -r /conf/conf.d/md ] && cp /conf/conf.d/md /conf/conf.d/md.orig
 					echo "MD_DEVS=$device " >> /conf/conf.d/md
-					/scripts/local-top/mdadm
-					mv /conf/conf.d/md.orig /conf/conf.d/md
+					/scripts/local-top/mdadm >>/boot.log
+					[ -r /conf/conf.d/md.orig ] && mv /conf/conf.d/md.orig /conf/conf.d/md
 				fi
 				;;
 		esac
