From: Evgeni Golov <evgeni@debian.org>
Subject: restore support for old persistence media

Index: live-boot-grml/components/9990-aaa-fixme.sh
===================================================================
--- live-boot-grml.orig/components/9990-aaa-fixme.sh	2014-03-08 13:43:19.172121783 +0100
+++ live-boot-grml/components/9990-aaa-fixme.sh	2014-03-08 13:43:19.164122722 +0100
@@ -26,5 +26,8 @@
 	export PATH
 fi
 
+old_root_overlay_label="live-rw"
+old_home_overlay_label="home-rw"
 custom_overlay_label="persistence"
+old_custom_overlay_label="custom-ov"
 persistence_list="persistence.conf"
Index: live-boot-grml/components/9990-misc-helpers.sh
===================================================================
--- live-boot-grml.orig/components/9990-misc-helpers.sh	2014-03-08 13:43:19.172121783 +0100
+++ live-boot-grml/components/9990-misc-helpers.sh	2014-03-08 13:43:19.168122252 +0100
@@ -1632,6 +1632,32 @@
 	echo ${used_devices}
 }
 
+fix_backwards_compatibility ()
+{
+	local device dir opt backing include_list
+	device=${1}
+	dir=${2}
+	opt=${3}
+
+	if [ -n "${PERSISTENCE_READONLY}" ]
+	then
+		return
+	fi
+
+	backing="$(mount_persistence_media ${device})"
+	if [ -z "${backing}" ]
+	then
+		return
+	fi
+
+	include_list="${backing}/${persistence_list}"
+	if [ ! -r "${include_list}" ] && [ ! -r "${backing}/${old_persistence_list}" ]
+	then
+		echo "# persistence backwards compatibility:
+${dir} ${opt},source=." > "${include_list}"
+	fi
+}
+
 is_mountpoint ()
 {
 	directory="$1"
Index: live-boot-grml/components/9990-overlay.sh
===================================================================
--- live-boot-grml.orig/components/9990-overlay.sh	2014-03-08 13:43:19.172121783 +0100
+++ live-boot-grml/components/9990-overlay.sh	2014-03-08 13:43:19.168122252 +0100
@@ -210,7 +210,7 @@
 
 		if is_in_comma_sep_list overlay ${PERSISTENCE_METHOD}
 		then
-			overlays="${custom_overlay_label}"
+			overlays="${old_root_overlay_label} ${old_home_overlay_label} ${custom_overlay_label} ${old_custom_overlay_label}"
 		fi
 
 		local overlay_devices
@@ -222,10 +222,27 @@
 				media="$(echo ${media} | tr ":" " ")"
 
 				case ${media} in
+					${old_root_overlay_label}=*)
+						device="${media#*=}"
+						fix_backwards_compatibility ${device} / union
+						overlay_devices="${overlay_devices} ${device}"
+						;;
+
+					${old_home_overlay_label}=*)
+						device="${media#*=}"
+						fix_backwards_compatibility ${device} /home bind
+						overlay_devices="${overlay_devices} ${device}"
+						;;
+
 					${custom_overlay_label}=*)
 						device="${media#*=}"
 						overlay_devices="${overlay_devices} ${device}"
 						;;
+
+					${old_custom_overlay_label}=*)
+						device="${media#*=}"
+						overlay_devices="${overlay_devices} ${device}"
+						;;
 				 esac
 			done
 		fi
